libs/s25main/ingameWindows/IngameWindow.cpp

```diff
@@ -45,8 +47,9 @@ IngameWindow::IngameWindow(unsigned id, const DrawPoint& pos, const Extent& size
     // Load last position or center the window
     if(pos == posLastOrCenter)
     {
-        if(id < last_pos.size() && last_pos[id].isValid())
-            SetPos(last_pos[id]);
+        auto& settings = SETTINGS.windows.persistentSettings;
+        if(helpers::contains(persistentWindows, GetID()) && settings[GetID()].lastPos.isValid())
+            SetPos(settings[GetID()].lastPos);
```

Comment:
does this check if the window location is "out of bounds"?

Reply:
SetPos does the out of bounds check, yes